privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -24,10 +25,28 @@ def __init__(
         :param dataset: dataset for training of shadow classifiers, test_data from dataset
         with concatenation [test_features, test_labels]
         """
+
+        if not (isinstance(dataset, tuple) and list(map(type, dataset)) == [np.ndarray, np.ndarray]):
+            raise TypeError("Dataset type should be of shape (np.ndarray, np.ndarray).")
+            
         self.dataset = dataset
-        # count of shadow training sets, must be eval
+
+        if not(isinstance(target_model, TensorFlowV2Classifier) or isinstance(target_model,PyTorchClassifier)):
+            raise TypeError("Target model must be of type Classifier.")
+
+        # count of shadow training sets, must be even
         self.amount_sets = 2
+        if self.amount_sets % 2 != 0 or self.amount_sets < 2:
+            raise ValueError("Number of shadow classifiers must be even and greater than 2.")
```

Comment:
ValueError message is not 100% correct: Number of shadow classifiers must be >= 2, not >2

Reply:
Should be correct because we want to raise an error when <2 amount_sets